frontend/lib/src/ForwardMessageCache.ts

```diff
@@ -92,24 +93,32 @@ export class ForwardMsgCache {
   ): Promise<ForwardMsg> {
     this.maybeCacheMessage(msg, encodedMsg)
 
-    if (msg.type !== "refHash") {
+    if (msg.type !== "forwardMsgRef") {
       return msg
     }
 
-    let newMsg = this.getCachedMessage(msg.refHash as string, true)
+    let newMsg = this.getCachedMessage(
+      msg.forwardMsgRef?.refHash as string,
+      true
+    )
     if (newMsg != null) {
-      logMessage(`Cached ForwardMsg HIT [hash=${msg.refHash}]`)
+      logMessage(`Cached ForwardMsg HIT [hash=${msg.forwardMsgRef?.refHash}]`)
     } else {
       // Cache miss: fetch from the server
-      logMessage(`Cached ForwardMsg MISS [hash=${msg.refHash}]`)
+      logMessage(`Cached ForwardMsg MISS [hash=${msg.forwardMsgRef?.refHash}]`)
       const encodedNewMsg = await this.endpoints.fetchCachedForwardMsg(
```

Comment:
If a (large) message gets fetched via the endpoint, should we rely on the backend implementation to also set the cacheable flag for large messages or should we always cache messages?